kiwi/xml_state.py

```diff
@@ -522,33 +522,52 @@ def get_users(self):
         """
         List of configured users.
 
-        Each entry in the list is a tuple with the following information
-
-        * group_name, name of the group
-        * group_id, id of the group
-        * user_sections, list of xml_parse::users instances which belong
-          to the group name and id of this tuple
+        Each entry in the list is a single xml_parse::user instance.
 
         :return: user data
         :rtype: list
         """
-        users_by_group_type = namedtuple(
-            'users_by_group_type', ['group_name', 'group_id', 'user_sections']
-        )
-        users_by_group = []
+        users_list = []
         users_sections = self.get_users_sections()
-        if users_sections:
-            for users in users_sections:
-                user_sections = users.get_user()
-                if user_sections:
-                    users_by_group.append(
-                        users_by_group_type(
-                            group_name=users.get_group(),
-                            group_id=users.get_id(),
-                            user_sections=user_sections
-                        )
-                    )
-        return users_by_group
+        if len(users_sections):
+            for users_section in users_sections:
+                for user in users_section.get_user():
+                    match = [i for i, u in enumerate(users_list) if u.get_name() == user.get_name()]
+                    for idx in match:
+                        if users_list[idx].get_groups() and user.get_groups():
+                            users_list[idx].set_groups(
+                                ','.join([users_list[idx].get_groups(), user.get_groups()])
+                            )
+                        elif not users_list[idx].get_groups() and user.get_groups():
+                            users_list[idx].set_groups(user.get_groups())
```

Comment:
I'm a little bit concerned that get_users changes the groups attribute, why is this needed ?